### PR TITLE
Bug 2091602: Support AWS GovCloud

### PR DESCRIPTION
**Pull request by alebedev87**
Main changes related to AWS Gov Cloud support:
- `pkg/operator/controller/externaldns/credentials_request.go#createProviderConfig()` - added [the AWS Gov partition](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/r53-api-permissions-ref.html) to the hosted zone ARN
- `pkg/operator/controller/externaldns/pod.go#fillAWSFileds()` - added `AWS_REGION` env variable as [required by the upstream](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/aws.md#govcloud-caveats)

The rest is the refactoring I could not pass by:
- `pkg/operator/controller/externaldns/credentials-request.go` renamed to `pkg/operator/controller/externaldns/credentials_request.go`
- `pkg/operator/controller/externaldns/credentials-request_test.go` renamed to `pkg/operator/controller/externaldns/credentials_request_test.go`
- `pkg/operator/controller/externaldns/credentials_request.go` - methods functions organized - `reconciler` methods up, secondary functions down
- `pkg/operator/controller/externaldns/credentials_request.go#externalDNSCredentialsRequestChanged()` - was just not good, wasn't comparing service accounts, secrets
- `pkg/operator/controller/externaldns/credentials_request_test.go` was basically rewritten, better to review it as a new file
- `pkg/operator/controller/utils/test/builders.go` was added to greatly simplify the generation of test `ExternalDNS` CRs

### CODE REVIEW

File: pkg/operator/controller/externaldns/credentials_request_test.go [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+		{
+			name:                      "Update drifted credentials request in Azure. Provider spec",
+			existingObjects:           []runtime.Object{makeCredentialsRequest("externaldns-credentials-request-azure").withSAs("external-dns-operator").withSecret("externaldns-cloud-credentials", "external-dns-operator").withProviderSpec(undesiredAzureProviderSpec).do()},
+			inputExtDNS:               test.MakeExternalDNS(test.Name).WithAzure().WithRouteSource().WithZones("public-zone").Do(),
+			expectedCredentialRequest: makeCredentialsRequest("externaldns-credentials-request-azure").withSAs("external-dns-operator").withSecret("externaldns-cloud-credentials", "external-dns-operator").withProviderSpec(desiredAzureProviderSpec).do(),
+		},
+		{
+			name:                      "Create credentials request from scratch in GCP",
+			existingObjects:           []runtime.Object{},
+			inputExtDNS:               test.MakeExternalDNS(test.Name).WithGCP().WithRouteSource().WithZones("public-zone").Do(),
```
Comment (thejasn) [inline pkg/operator/controller/externaldns/credentials_request_test.go:88]:
```suggestion
			inputExtDNS:               test.NewExternalDNS(test.Name).WithGCP().WithRouteSource().WithZones("public-zone").Build(),
```

File: pkg/operator/controller/externaldns/credentials_request_test.go [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+			expectedCredentialRequest: makeCredentialsRequest("externaldns-credentials-request-gcp").withSAs("external-dns-operator").withSecret("externaldns-cloud-credentials", "external-dns-operator").withProviderSpec(desiredGCPProviderSpec).do(),
```
Comment (thejasn) [inline pkg/operator/controller/externaldns/credentials_request_test.go:89]:
```suggestion
			expectedCredentialRequest: newCredentialsRequest("externaldns-credentials-request-gcp").withSAs("external-dns-operator").withSecret("externaldns-cloud-credentials", "external-dns-operator").withProviderSpec(desiredGCPProviderSpec).build(),
```

**Review by thejasn — COMMENTED** [review]
Nit: Better to prefer convention.

File: pkg/operator/controller/externaldns/credentials_request_test.go (hunk 1 above)
Comment (alebedev87) [reply pkg/operator/controller/externaldns/credentials_request_test.go]: Done

File: pkg/operator/controller/externaldns/credentials_request_test.go (hunk 2 above)
Comment (alebedev87) [reply pkg/operator/controller/externaldns/credentials_request_test.go]: Done

**Comment by quarterpin** [comment]
/label qe-approved
Verified via pre-merge test process. For more details refer to: https://bugzilla.redhat.com/show_bug.cgi?id=2091602#c2

**Comment by alebedev87** [comment]
`main` moved from the time the last run was made, so after all the labels were given the run failed due to the changes made in between:
```
- 	SecurityContext: nil,
+ 	SecurityContext: s"&SecurityContext{Capabilities:&Capabilities{Add:[],Drop:[ALL],},Privileged:*false,SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:*true,ReadOnlyRootFilesystem:nil,AllowPrivilegeEscalation:*false,RunAsGroup:nil,ProcMount:nil,WindowsOptions:nil,SeccompProfile:"...,
```

Rebased and added the missing security context to the unit test. No functional changes - all the labels can be set back.

**Pull request merged**